handlers/jiochat/jiochat.go

```diff
@@ -0,0 +1,323 @@
+package jiochat
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha1"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"github.com/buger/jsonparser"
+	"github.com/garyburd/redigo/redis"
+	"github.com/nyaruka/courier"
+	"github.com/nyaruka/courier/handlers"
+	"github.com/nyaruka/courier/utils"
+	"github.com/nyaruka/gocommon/urns"
+	"net/http"
+	"net/url"
+	"sort"
+	"strings"
+	"time"
+)
+
+var maxMsgLength = 1600
+var jiochatFetchTimeout = time.Second * 2
+
+const configJiochatAppID = "jiochat_app_id"
+const configJiochatAppSecret = "jiochat_app_secret"
+
+var jiochatURL = "https://channels.jiochat.com"
+
+func init() {
+	courier.RegisterHandler(newHandler(&jiochatClient{}))
+}
+
+type handler struct {
+	handlers.BaseHandler
+	jiochatClient jiochatAccess
```

Comment:
I don't understand the purpose of this client / interface? Why have it?

Reply:
the interface allow me to mock the redis and be able to test the header is properly tested

Reply:
Why do you need that vs a mock server to satisfy these? (and a configurable URL that they hit?)

Reply:
I am unable to set values on Redis on the mock server